api/validator_test.go

```diff
@@ -110,6 +110,382 @@ var _ = Describe("Validator", func() {
 	JustBeforeEach(func() {
 		err = validator.Validate()
 	})
+
+	Describe("validating top-level spec", func() {
```

Comment:
in most of your cases, I think you could probably do a describeTable instead? I _think_ it would take less lines to achieve the same result.